component/prometheus/operator/podmonitors/config_gen.go

```diff
@@ -0,0 +1,176 @@
+package podmonitors
+
+// SEE https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg.go
```

Comment:
in general, I would like to see more tests for this. For example, some of the code is adapted from prometheus (as referenced in the link), so there are tests we could already reuse https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/promcfg_test.go